flagsmith/flagsmith.py

```diff
@@ -238,6 +229,9 @@ def get_identity_flags(
         self,
         identifier: str,
         traits: typing.Optional[typing.Mapping[str, TraitValue]] = None,
+        *,
+        transient: bool = False,
+        transient_traits: typing.Optional[typing.List[str]] = None,
```

Comment:
> I don't necessarily disagree with the approach here, but my original expectation was that we would change the interface to pass traits as a list of e.g. Trait objects that have a transient attribute.

This is a breaking change, though. Is it worth it for such a minor feature?

> If we're going to keep this as just a list of keys, I would like to updating the argument name to `transient_trait_keys`.

Agreed, will do.

Reply:
This is done.

Reply:
>This is a breaking change, though. Is it worth it for such a minor feature?

Well, no, but the interface here is a little ugly don't you think? I'd certainly like to, at some point, improve the interface. If it was just python that we were worrying about, I'd suggest that we could just accept 2 different data types for the `traits` argument (the current mapping, and a list of new `Trait` objects) and then raise a warning that using the mapping is to be deprecated. 

I wonder if this approach would be possible in all our SDKs using overrides?

Reply:
As discussed offline, I've introduced an optional `TraitConfig` typed dict that can be provided instead of a trait value and includes a `"transient"` key.